plugins/woocommerce-admin/client/marketing/data-multichannel/actions.ts

```diff
@@ -38,9 +43,41 @@ export const receiveRecommendedChannelsError = ( error: ApiFetchError ) => {
 	};
 };
 
+export const receiveCampaignsSuccess = ( success: {
+	payload: Array< Campaign >;
+	error: boolean;
+	meta: {
+		page: number;
+		perPage: number;
+		total?: number;
+	};
+} ) => {
+	return {
+		type: TYPES.RECEIVE_CAMPAIGNS_SUCCESS,
+		...success,
+	};
+};
+
+export const receiveCampaignsError = ( error: {
+	payload: ApiFetchError;
+	error: boolean;
```

Comment:
💅  Is not `error` field a bit redundant taking in consideration we have an specific error method? I take as granted that in `receiveCampaignsError `, `error` field will be always true.

Reply:
💅 Another improvement in this types could be to create an interface or type definition ie `ReceiveCampaigns`

```js
    payload: Array< Campaign > | ApiFetchError;
	meta: {
		page: number;
		perPage: number;
		total?: number;
	};
```

Or even a single method 

```js
export const receiveCampaigns = ( response: {
	payload: Array< Campaign > | ApiFetchError;
	error: boolean;
	meta: {
		page: number;
		perPage: number;
		total?: number;
	};
} ) => {
	return {
		type: error ? TYPES.RECEIVE_CAMPAIGNS_ERROR : RECEIVE_CAMPAIGNS_SUCCESS,
		...response,
	};
};
```

Going beyond we can even have a single type in the reducer. 


```js
export const receiveCampaigns = ( response: {
	payload: Array< Campaign > | ApiFetchError;
	error: boolean;
	meta: {
		page: number;
		perPage: number;
		total?: number;
	};
} ) => {
	return {
		type: TYPES.RECEIVE_CAMPAIGNS,
		...response,
	};
};
```

Just needs a little tweak in the reducer.

```js
[ action.meta.page ]: error ? { error: action.payload, } :  {
							data: action.payload,
						},
```

Reply:
> 💅 Is not `error` field a bit redundant taking in consideration we have an specific error method? In `receiveCampaignsError `, `error` field will be always true. And in `receiveCampaignsSuccess ` error will be always `false`.

You're right, I just realized that. 😂 

> 💅 Another improvement in this types could be to create an interface or type definition ie `ReceiveCampaigns`
> Or even a single method
> Going beyond we can even have a single type in the reducer.

That is actually what I tried to do but I couldn't make it work. Thanks for the code examples. I used your code and incorporated [discriminated unions](https://www.typescriptlang.org/docs/handbook/2/narrowing.html#discriminated-unions) in [`9feaf55` (#36735)](https://github.com/woocommerce/woocommerce/pull/36735/commits/9feaf55a6070e11ceb8633d38f22e1b51fa645c9) and now it works nicely. 😎 🤜 🤛